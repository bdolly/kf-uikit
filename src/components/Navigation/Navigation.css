.Navigation {
    @apply
        flex
        bg-white
        h-24
        border-b-2
        p-4;
}

.Navigation .Button {
    @apply
        mx-2
}
