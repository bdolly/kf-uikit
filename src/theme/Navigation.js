const navigation = {
  navLink: css`
    display: block;
    color: ${colors.hover};
    padding: 6px 10px;
    margin: 0px 4px;
    text-decoration: none;
    border: solid 2px transparent;
    font-size: 14px;
    font-weight: 500;

    &:hover {
      ${mixins.linkButtonActive};
    }
  `,
  navBar: css`
    list-style-type: none;
    margin: 0;
    padding: 0;
    display: flex;
    flex-direction: row;
    align-items: center;
    font-size: 14px;
    line-height: 1.86;
    letter-spacing: 0.2px;
    font-size: 14px;
    font-weight: 500;
    line-height: 1.86;
    letter-spacing: 0.2px;
    text-align: left;
    color: #90278e;
  `,

  secondaryNav: css`
    list-style-type: none;
    margin: 0;
    padding: 0;
    overflow: hidden;
    display: flex;
    flex-direction: row;
    align-items: center;
    line-height: 1.86;
    letter-spacing: 0.2px;
    border-left: 1px solid ${colors.greyScale5};

    li {
      box-sizing: content-box;
      border-right: 1px solid ${colors.greyScale5};
      a {
        display: block;
        font-size: 16px;
        line-height: 1.86;
        letter-spacing: 0.2px;
        text-align: left;
        color: ${colors.primary};
        font-weight: 500;
        padding: 10px 40px;
        border-bottom: 5px solid transparent;
        text-decoration: none;
      }

      a:hover,
      a.active {
        cursor: pointer;
        color: ${colors.highlight};
        font-weight: 500;
        border-bottom: 5px solid ${colors.highlight};
        text-decoration: none;
      }
    }
  `
};
export default navigation;
