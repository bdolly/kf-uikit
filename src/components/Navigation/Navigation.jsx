import React from 'react';
import propTypes from 'prop-types';
import className from 'classnames';
import Logo from '../Logo/Logo';
import './Navigation.css';

/**
 * The common Kids First navigation.
 * Must specify button actions to use
 */
const Navigation = ({ buttons, children }) => {

  const navClass = className(
    'Navigation',
  );

  return (
    <nav className={navClass}>
      <Logo />
      <div className='flex items-center justify-start w-full'>
        {buttons.map(button => button)}
      </div>
      {children}
    </nav>
  );
}

Navigation.propTypes = {
  /** Buttons to display on the navbar */
  buttons: propTypes.node,
  /** Child elements to display on the navbar */
  children: propTypes.node,
};

Navigation.defaultProps = {
  buttons: null,
  children: null,
};

export default Navigation;
