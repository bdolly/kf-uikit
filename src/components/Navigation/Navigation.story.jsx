/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs';
import Button from '../Button/Button';
import Navigation from './Navigation';

const stories = storiesOf('Navigation', module);

stories.addDecorator(withKnobs);

stories.add('Header Navigation', () => (
  <Navigation
    buttons={[
      <Button>Dashboard</Button>,
      <Button outline>File Repository</Button>]}
  />
));
