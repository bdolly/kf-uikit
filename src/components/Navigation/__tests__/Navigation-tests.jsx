import React from 'react';
import renderer from 'react-test-renderer';
import Navigation from '../Navigation';

it(`navigation renders correctly`, () => {
    const tree = renderer.create(
      <Navigation />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
