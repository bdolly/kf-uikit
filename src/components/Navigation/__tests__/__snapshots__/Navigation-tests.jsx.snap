// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navigation renders correctly 1`] = `
<nav
  className="Navigation"
/>
`;
